tests/test_aggregates.py

```diff
@@ -0,0 +1,253 @@
+        def helper_default_aggregator_function(jobs):
+            return [jobs]
+
+        def helper_non_default_aggregator_function(jobs):
+            for job in jobs:
+                yield (job,)
+
+        list_of_aggregates = [
+            aggregator(), aggregator(), aggregator(helper_default_aggregator_function),
+            aggregator(helper_non_default_aggregator_function),
+            aggregator(helper_non_default_aggregator_function),
+            aggregator.groupsof(1), aggregator.groupsof(1),
+            aggregator.groupsof(2), aggregator.groupsof(3), aggregator.groupsof(4),
+            aggregator.groupby('half'), aggregator.groupby('half'), aggregator.groupby('even'),
+            aggregator.groupby('half', -1), aggregator.groupby('even', -1),
+            aggregator.groupby(['half', 'even'], default=[-1, -1])
+        ]
+
+        assert len(set(list_of_aggregates)) == 11
+
+
+# Test the _AggregatesStore and _DefaultAggregateStore classes
+class TestAggregateStoring(AggregateProjectSetup):
+
+    def test_valid_aggregator_function(self, setUp, project):
+        # Return groups of 1
+        def helper_aggregator_function(jobs):
+            for job in jobs:
+                yield (job,)
+
+        aggregate_instance = aggregator(helper_aggregator_function)
+        aggregate_instance = aggregate_instance._create_AggregatesStore(project)
+
+        aggregate_job_manual = helper_aggregator_function(project)
+
+        assert [aggregate for aggregate in aggregate_job_manual] == \
+               list(aggregate_instance)
+
+    def test_valid_lambda_aggregator_function(self, setUp, project):
+        aggregate_instance = aggregator(lambda jobs: [jobs])
+        aggregate_instance = aggregate_instance._create_AggregatesStore(project)
+
+        assert [tuple(project)] == list(aggregate_instance)
+
+    def test_valid_sort_by(self, setUp, project):
+        helper_sort = partial(sorted, key=lambda job: job.sp.i)
+        aggregate_instance = aggregator(sort_by='i')
+        aggregate_instance = aggregate_instance._create_AggregatesStore(project)
+
+        assert [tuple(helper_sort(project))] == list(aggregate_instance)
+
+    def test_valid_reversed_sort(self, setUp, project):
+        helper_sort = partial(sorted, key=lambda job: job.sp.i, reverse=True)
+        aggregate_instance = aggregator(sort_by='i', reverse_order=True)
+        aggregate_instance = aggregate_instance._create_AggregatesStore(project)
+        assert [tuple(helper_sort(project))] == list(aggregate_instance)
+
+    def test_groups_of_valid_num(self, setUp, project):
+        valid_values = [1, 2, 3, 6, 10]
+        expected_aggregates = [10, 5, 4, 2, 1]
+
+        for i, valid_value in enumerate(valid_values):
+            aggregate_instance = aggregator.groupsof(valid_value)
+            aggregate_instance = aggregate_instance._create_AggregatesStore(project)
+            assert len(aggregate_instance) == expected_aggregates[i]
```

Comment:
This does not test that each aggregate of `aggegrator.groupsof(2)` for instance has two jobs (except the last perhaps).